src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerExecutor.cs

```diff
@@ -792,75 +870,13 @@ private void ExecuteBlockActionsCore<TBlockStartAction, TBlockAction, TNodeActio
 
             // Include the stateful actions.
             foreach (var startAction in startActions)
-            {
-                if (startAction is CodeBlockStartAnalyzerAction<TLanguageKindEnum> codeBlockStartAction)
```

Comment:
we'd do runtime checks to then have syntaxnode-vs-operation specific behavior.  now, we just pass the data we need to the 'addActions' function, and each caller does the specific logic they want in a properly strongly typed fashion.